### PR TITLE
Support DYNAMIC_ARCH with CMake

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: kernel/CMakeLists.txt [hunk 1]
```diff
@@ -246,6 +246,23 @@ function (build_core TARGET_CORE KDIR TSUFFIX KERNEL_DEFINITIONS)
...
+          GenerateNamedObjects("generic/z${name}_lcopy_${${float_char}GEMM3M_UNROLL_M}.c" "USE_ALPHA;IMAGE_ONLY" "${name}_olcopyi.c" false "" "" false ${float_type})
+          GenerateNamedObjects("generic/z${name}_ucopy_${${float_char}GEMM3M_UNROLL_M}.c" "" "${name}_iucopyb.c" false "" "" false ${float_type})
+          GenerateNamedObjects("generic/z${name}_ucopy_${${float_char}GEMM3M_UNROLL_M}.c" "REAL_ONLY" "${name}_iucopyr.c" false "" "" false ${float_type})
+          GenerateNamedObjects("generic/z${name}_ucopy_${${float_char}GEMM3M_UNROLL_M}.c" "IMAGE_ONLY" "${name}_iucopyi.c" false "" "" false ${float_type})
+          GenerateNamedObjects("generic/z${name}_lcopy_${${float_char}GEMM3M_UNROLL_M}.c" "" "${name}_ilcopyb.c" false "" "" false ${float_type})
+          GenerateNamedObjects("generic/z${name}_lcopy_${${float_char}GEMM3M_UNROLL_M}.c" "REAL_ONLY" "${name}_ilcopyr.c" false "" "" false ${float_type})
+          GenerateNamedObjects("generic/z${name}_lcopy_${${float_char}GEMM3M_UNROLL_M}.c" "IMAGE_ONLY" "${name}_ilcopyi.c" false "" "" false ${float_type})
```
Comment (isuruf) [inline kernel/CMakeLists.txt:266]: I added these because they were missing. Needs a careful look to see if it is correct.

**Comment by jschueller** [comment]
maybe DYNAMIC_ARCH could be defined as a cmake option so it appears in cache, ccmake etc

**Comment by isuruf** [comment]
Any ideas why this fails with symbols missing? (On linux, the build goes fine probably because gcc exports all symbols.)

**Comment by martin-frbg** [comment]
Any chance to access the static library that was built before this point and see what symbols it contains ?

**Comment by isuruf** [comment]
Static libraries are not built on Windows with MSVC

**Comment by isuruf** [comment]
It seems that the problem is that `openblas.def` defines `dcabs1_`, but when DYNAMIC_ARCH=1, there's no `dcabs1_` but multiple symbols with the arch as a suffix `dcabs1_ATOM_`.
When an executable links to the dll using the import library `openblas.lib` it will complain that `dcabs1_` is missing even though it's not used.

**Comment by martin-frbg** [comment]
~~Sorry, I am not familiar with the MSVC build process. Guess it is the openblas.def file then that could be relevant ?~~ Sorry again, our messages crossed (but good to know I made the right guess). I seem to remember there is one catchall dcabs1_ that forwards to dcabs1_ATOM_ etc generated "somewhere" (defined in a header file I think ?) in the Linux DYNAMIC_ARCH build, maybe that just needs adjusting an #ifdef to handle clang/msvc as well.
EDIT: just ignore me for now :-)

**Comment by isuruf** [comment]
I think I know the issue. In `kernel/Makefile` the symbols which are missing `lsame`, `scabs1` and `dcabs1` are marked as `COMMONOBJS`. In the CMake build process, they are not marked as `COMMONOBJS` and built for each architecture with the architecture name appended. I'll fix this.

**Comment by isuruf** [comment]
Appveyor passes. Travis job got killed again.

**Pull request merged**